src/args.jl

```diff
@@ -1,1649 +1,1665 @@
-        plotattributes[csym] = if plotattributes[csym] == :auto
-            plot_color(if has_black_border_for_default(stype) && s == :line
-                sp[:foreground_color_subplot]
-            else
-                scolor
-            end)
-        elseif plotattributes[csym] == :match
-            plot_color(scolor)
-        else
-            get_series_color(plotattributes[csym], sp, plotIndex, stype)
-        end
-    end
-
-    # update markerstrokecolor
-    plotattributes[:markerstrokecolor] = if plotattributes[:markerstrokecolor] == :match
-        plot_color(sp[:foreground_color_subplot])
-    elseif plotattributes[:markerstrokecolor] == :auto
-        get_series_color(plotattributes[:markercolor], sp, plotIndex, stype)
-    else
-        get_series_color(plotattributes[:markerstrokecolor], sp, plotIndex, stype)
-    end
-
-    # if marker_z, fill_z or line_z are set, ensure we have a gradient
-    if plotattributes[:marker_z] !== nothing
-        ensure_gradient!(plotattributes, :markercolor, :markeralpha)
-    end
-    if plotattributes[:line_z] !== nothing
-        ensure_gradient!(plotattributes, :linecolor, :linealpha)
-    end
-    if plotattributes[:fill_z] !== nothing
-        ensure_gradient!(plotattributes, :fillcolor, :fillalpha)
-    end
-
-    # scatter plots don't have a line, but must have a shape
-    if plotattributes[:seriestype] in (:scatter, :scatterbins, :scatterhist, :scatter3d)
-        plotattributes[:linewidth] = 0
-        if plotattributes[:markershape] == :none
-            plotattributes[:markershape] = :circle
-        end
-    end
-
-    # set label
-    label = plotattributes[:label]
-    label = (label == "AUTO" ? "y$globalIndex" : label)
-    label = label in (:none, nothing, false) ? "" : label
-    plotattributes[:label] = label
-
-    _replace_linewidth(plotattributes)
-   plotattributes
-end
-
-function _series_index(plotattributes, sp)
-    idx = 0
-    for series in series_list(sp)
-        if series[:primary]
-            idx += 1
-        end
-        if series == plotattributes
-            return idx
-        end
-    end
-    if get(plotattributes, :primary, true)
-        idx += 1
-    end
-    return idx
-end
+const _keyAliases = Dict{Symbol,Symbol}()
+
+function add_aliases(sym::Symbol, aliases::Symbol...)
+    for alias in aliases
+        if haskey(_keyAliases, alias)
+            error("Already an alias $alias => $(_keyAliases[alias])... can't also alias $sym")
+        end
+        _keyAliases[alias] = sym
+    end
+end
+
+function add_non_underscore_aliases!(aliases::Dict{Symbol,Symbol})
+    for (k,v) in aliases
+        s = string(k)
+        if '_' in s
+            aliases[Symbol(replace(s, "_" => ""))] = v
+        end
+    end
+end
+
+function add_non_underscore_aliases!(aliases::Dict{Symbol,Symbol}, args::Vector{Symbol})
+    for arg in args
+        s = string(arg)
+        if '_' in s
+            aliases[Symbol(replace(s, "_" => ""))] = arg
+        end
+    end
+end

```

Comment:
This was the only code I added. Even I am quite surprised with these large differences.